packages/search/src/common/params.ts

```diff
@@ -0,0 +1,27 @@
+/* Copyright (c) 2017 Environmental Systems Research Institute, Inc.
+* Apache-2.0 */
+
+export interface ISearchParams {
```

Comment:
we actually consider it an anti-pattern to extend [`IParams`](https://esri.github.io/arcgis-rest-js/api/request/IParams). 

see https://github.com/Esri/arcgis-rest-js/pull/78 for more info.

Reply:
I am not sure I understand this correctly. I am not extending `IParams` here. So it is good, right?

Reply:
> I am not extending IParams

you've got me there.

the point is that you want your new method needs to accept an instance of `IRequestOptions` to pass through to `request`.

this means that you'd be better served to extend `IRequestOptions` with new custom props and pass them down internally yourself.

check out pat's example here: https://github.com/Esri/arcgis-rest-js/pull/78#issuecomment-352124790